### PR TITLE
[RFC] Stage 0 - Add new cloud fields

### PR DESCRIPTION
**Pull request by legoguy1000**
- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)? yes
- Have you followed the [contributor guidelines](https://github.com/elastic/ecs/blob/main/CONTRIBUTING.md)? yes
- For proposing substantial changes or additions to the schema, have you reviewed the [RFC process](https://github.com/elastic/ecs/blob/main/rfcs/README.md)? yes
- If submitting code/script changes, have you verified all tests pass locally using `make test`?
- If submitting schema/fields updates, have you generated new artifacts by running `make` and committed those changes?
- Is your pull request against main? Unless there is a good reason otherwise, we prefer pull requests against main and will backport as needed.
- Have you added an entry to the [CHANGELOG.next.md](https://github.com/elastic/ecs/blob/main/CHANGELOG.next.md)?

### CODE REVIEW

**Comment by djptek** [comment]
Hi @legoguy1000 this looks really interesting. Do you have examples of e.g. GCP or Azure logs that could be included alongside the AWS logs to illustrate the impartiality of the mapping?

**Comment by legoguy1000** [comment]
GCP Flow logs:
..."src_vpc":{"project_id":"my-sample-project","subnetwork_name":"default",**"vpc_name":"default"**}...

GCP DNS logs
..."sourceNetwork":"default",...

**Comment by legoguy1000** [comment]
I added some more examples to the RFC as well as 2 more additional fields for `subnetwork` name and ID just like VPC.  While I may not have more examples at this time, i still think its impartial as every major cloud service has these terms (or equiv) so even though i don't have specific logs that include them, its not like the data doesn't exist.

**Comment by djptek** [comment]
linking [related comment in Add new cloud.* (or cloud related fields) #1725](https://github.com/elastic/ecs/issues/1725#issuecomment-1156025912)

File: rfcs/0000-new-cloud-fields.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+Stage X: Provide a brief explanation of why the proposal is being marked as abandoned. This is useful context for anyone revisiting this proposal or considering similar changes later on.
+-->
+
+## Fields
+
+Initial list of proposed fields:
+
+* `cloud.edge_location`: The edge location, usually an airport IATA code (DFW), of a cloud service. Very common for DNS, CDN, or other cloud services
+* `cloud.instance.image.id`: ID of the image used to create the virtual instance/machine
+* `cloud.instance.image.name`: Name of the image used to create teh virtual instance/machine
```
Comment (ebeahan) [inline rfcs/0000-new-cloud-fields.md:32]:
```suggestion
* `cloud.instance.image.name`: Name of the image used to create the virtual instance/machine
```

File: rfcs/0000-new-cloud-fields.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+* `cloud.vpc.name`: Name of the virtual network
+* `cloud.vpc.id`: ID of the virtual network
+* `cloud.subnet.name`: Name of the Subnet within the VPC/Virtual network
+* `cloud.subnet.id`: ID of the Subnet within the VPC/Virtual network
+* `cloud.instance.lifecycle`: Type of instance *normal* vs *spot*, see https://github.com/elastic/ecs/issues/323 for more discussion
+* `cloud.object_store.name`: Name of S3/Compatible storage
```
Comment (ebeahan) [inline rfcs/0000-new-cloud-fields.md:38]: I'm unsure about placing `object_store.*` fields underneath `cloud.*`. There are object store solutions like [MinIO](https://min.io/) and [ceph](https://docs.ceph.com/en/quincy/) which wouldn't automatically fit underneath `cloud.*`

Not a blocker for merging the proposal as-is for stage 0, but a topic to discuss and unpack more in a later stage.

**Review by ebeahan — APPROVED** [review]
Thanks for getting this RFC started, @legoguy1000!

The only [criteria](https://elastic.github.io/ecs/stages.html) for advancing to stage 0 is for proposed changes to appear useful and appropriate to ECS, and the changes suggested here are both.  

I noted a few small items. Once addressed, I think this is ready to merge.

**Comment by legoguy1000** [comment]
@ebeahan comments addressed

**Comment by legoguy1000** [comment]
@ebeahan Ae we good to merge this?

**Comment by ebeahan** [comment]
@legoguy1000 Yes - i set the RFC number and date and will merge.

**Pull request merged**